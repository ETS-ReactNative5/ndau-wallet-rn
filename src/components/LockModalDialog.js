import React, { Component } from 'react'

import { StyleSheet, Text } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import ModalDialog from './ModalDialog'
import CommonButton from '../components/CommonButton'
import { LockTransaction } from '../transactions/LockTransaction'
import { Transaction } from '../transactions/Transaction'

class LockModalDialog extends Component {
  constructor (props) {
    super(props)

    this.state = {
      period: '3m'
    }
  }

  setWallet = wallet => {
    this._wallet = wallet
  }

  setAccount = account => {
    this._account = account
  }

  _lock = async () => {
    this.props.startSpinner()
    this.closeModal()

    try {
      if (!this._wallet && !this._account) {
        console.warn('wallet and account are falsey in lock and should not be')
      } else {
        Object.assign(LockTransaction.prototype, Transaction)
        const lockTransaction = new LockTransaction(
          this._wallet,
          this._account,
          this.state.period
        )
        await lockTransaction.createSignPrevalidateSubmit()
      }
    } catch (error) {
      console.warn(error)
    }

    this.props.stopSpinner()
    // this.props.refresh()
  }

  showModal = () => {
    this._modalDialog.showModal()
  }

  closeModal = () => {
    this._modalDialog.closeModal()
  }

  render () {
    return (
      <ModalDialog
        ref={component => (this._modalDialog = component)}
        {...this.props}
      >
        <Text style={styles.text}>This is lock dialog</Text>
        <CommonButton title='Lock' onPress={this._lock} />
      </ModalDialog>
    )
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular',
    marginTop: hp('1%'),
    marginBottom: hp('1%'),
    marginLeft: wp('1%'),
    marginRight: wp('1%')
  }
})

export default LockModalDialog
