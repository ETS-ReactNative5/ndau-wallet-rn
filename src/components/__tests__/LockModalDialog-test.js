import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import LockModalDialog from '../LockModalDialog';

import renderer from 'react-test-renderer';

describe('testing LockModalDialog...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<LockModalDialog parentStyles={styles} navigator={navigator} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
