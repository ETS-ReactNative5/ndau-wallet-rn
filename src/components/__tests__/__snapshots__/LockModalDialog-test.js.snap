// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CommonButton... renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityTraits="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#4e957a",
      "borderRadius": 3,
      "height": 93.5,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
/>
`;

exports[`testing LockModalDialog... renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  navigator={
    Object {
      "setStyle": [Function],
    }
  }
  onRequestClose={[Function]}
  parentStyles={
    Object {
      "wizardText": Object {
        "color": "#ffffff",
        "fontSize": 20,
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#333333",
            "borderRadius": 3,
            "height": 934,
            "paddingBottom": 26.5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": 675,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#A9A9A9",
                "fontSize": 20,
              },
              Object {
                "marginLeft": 7.5,
                "marginTop": 6.5,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ÔÄç
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 20,
            "marginBottom": 13.5,
            "marginLeft": 7.5,
            "marginRight": 7.5,
            "marginTop": 13.5,
          }
        }
      >
        This is lock dialog
      </Text>
    </View>
  </View>
</Modal>
`;
